### PR TITLE
feat: implement URL rewriting for artifact downloads with proper URL parsing

### PR DESCRIPTION
**Pull request by eaugene**
This commit implements URL rewriting functionality for artifact downloads to support custom Buildkite API base URLs (e.g., proxy configurations) using proper URL parsing instead of string manipulation.

Changes Made:

Enhanced BuildkiteClientAdapter.DownloadArtifactByURL()
- Added URL rewriting logic before downloading artifacts
- Calls new rewriteArtifactURL() method to handle URL transformation

Added rewriteArtifactURL() method with proper URL parsing
- Uses url.Parse() to parse both input and base URLs
- Compares host and scheme components directly
- Rewrites URLs when base URL differs from input URL
- Handles base URLs with path prefixes and trailing slashes
- Generic implementation without hardcoded domain assumptions
- Gracefully handles malformed URLs and nil/empty base URLs

Improved error handling
- Enhanced error message in GetArtifact to include more context
- Added "response failed with error" prefix for better debugging

Added comprehensive unit tests
- TestBuildkiteClientAdapter_URLRewriting with 8 test cases:
  * URL rewriting when base URL has different host
  * No rewriting when host and scheme match exactly  
  * Rewriting for any domain (not just api.buildkite.com)
  * Base URLs without trailing slash handling
  * Scheme differences (http vs https)
  * Complex path prefix handling
  * Malformed URL handling
- TestBuildkiteClientAdapter_URLRewritingEdgeCases with 3 edge cases:
  * Nil base URL handling
  * Empty base URL handling  
  * Root path handling

Technical Details:
- Uses url.Parse() and URL component replacement for robust URL manipulation
- Avoids string manipulation to prevent URL parsing issues
- Generic host/scheme comparison without domain assumptions
- Preserves original URL structure while updating host, scheme, and path prefix
- Backwards compatible - no breaking changes to existing functionality

Test Plan:
✅ All existing tests continue to pass (100% test suite success)
✅ New URL rewriting tests pass with 11 scenarios covered
✅ Edge case handling verified with 3 additional test cases
✅ Comprehensive coverage of URL parsing edge cases
✅ Validated proper handling of various base URL configurations

This change enables the Buildkite MCP server to work correctly in environments where Buildkite API access goes through a proxy or custom gateway, ensuring artifact downloads use the same base URL as other API calls while using robust URL parsing techniques.

### CODE REVIEW

File: internal/buildkite/artifacts.go [hunk 1]
```diff
@@ -32,9 +33,29 @@ func (a *BuildkiteClientAdapter) ListByBuild(ctx context.Context, org, pipelineS
...
-	return a.Artifacts.DownloadArtifactByURL(ctx, url, writer)
+	// Rewrite URL if it's using the default Buildkite API URL and we have a custom base URL
+	rewrittenURL := a.rewriteArtifactURL(url)
+	return a.Artifacts.DownloadArtifactByURL(ctx, rewrittenURL, writer)
+}
+
+// rewriteArtifactURL rewrites artifact URLs to use the configured base URL
+func (a *BuildkiteClientAdapter) rewriteArtifactURL(url string) string {
+	// Get the configured base URL from the client
+	clientBaseURL := a.BaseURL.String()
```
Comment (wolfeidau) [inline internal/buildkite/artifacts.go:46]: Not a fan of doing .String() on the URL, then doing a string replacement.

Would prefer updating the url.Host with whatever the hostname is, or parsing the baseURL, and replacing Host and Scheme, this avoids the assumption of domain name.

**Review by wolfeidau — CHANGES_REQUESTED** [review]
Overall looks good, I understand the requirement and aside from the URL rewritting the change looks good.

File: internal/buildkite/artifacts.go (hunk 1 above)
Comment (eaugene) [reply internal/buildkite/artifacts.go]: Thanks for the review @wolfeidau  . resolved it

**Review by wolfeidau — APPROVED** [review]
Looks good
